compiler/test/suites/linking.re

```diff
@@ -16,4 +16,46 @@ describe("linking", ({test}) => {
     {|import List from "list"; print(List.map(n => n + 1, [1, 2, 3]))|},
     "[2, 3, 4]\n",
   );
+
+  test("no_start_section", ({expect}) => {
+    let name = "no_start_section";
+    let outfile = wasmfile(name);
+    compile(name, {|print("Hello, world!")|});
+    let sections =
+      Grain_utils.Wasm_utils.get_wasm_sections(open_in_bin(outfile));
+    let start_section =
+      List.find_opt(
+        (sec: Grain_utils.Wasm_utils.wasm_bin_section) =>
+          switch (sec) {
+          | {sec_type: Start} => true
+          | _ => false
+          },
+        sections,
+      );
+    //TODO: Check that _start is exported, waiting on Oscar's Wasm_utils update
+    expect.option(start_section).toBeNone();
+  });
+
+  test("use_start_section", ({expect}) => {
+    let name = "use_start_section";
+    let outfile = wasmfile(name);
+    compile(
+      ~config_fn=() => {Grain_utils.Config.use_start_section := true},
+      name,
+      {|print("Hello, world!")|},
+    );
+    let sections =
+      Grain_utils.Wasm_utils.get_wasm_sections(open_in_bin(outfile));
+    let start_section =
+      List.find_opt(
+        (sec: Grain_utils.Wasm_utils.wasm_bin_section) =>
+          switch (sec) {
+          | {sec_type: Start} => true
+          | _ => false
+          },
+        sections,
+      );
+    //TODO: Check that _start is not exported, waiting on Oscar's Wasm_utils update
```

Comment:
Maybe we should link this to an issue?

Reply:
I'm hoping to have it done before I merge this 🤞 - otherwise I'll open an issue.